salt/minion.py

```diff
@@ -828,12 +830,17 @@ def eval_master(self,
                 log.error(msg)
                 sys.exit(salt.defaults.exitcodes.EX_GENERIC)
 
+        # if the master_active_list is empty, reset with the original list
         # if we have a list of masters, loop through them and be
         # happy with the first one that allows us to connect
-        if isinstance(opts['master'], list):
+        if not opts['master_active_list']:
```

Comment:
We need to handle the case where this is not set yet. We could probably just make this a `get()` and be OK here.